packages/compiler/src/init/core-templates.ts

```diff
@@ -1,12 +1,22 @@
-import { readFile } from "fs/promises";
 import { CompilerPackageRoot } from "../core/node-host.js";
 import { resolvePath } from "../core/path-utils.js";
+import { CompilerHost } from "../index.js";
 
 export const templatesDir = resolvePath(CompilerPackageRoot, "templates");
+export interface LoadedCoreTemplates {
+  readonly baseUri: string;
+  readonly templates: Record<string, any>;
+}
 
-const content = JSON.parse(await readFile(resolvePath(templatesDir, "scaffolding.json"), "utf-8"));
-
-export const TypeSpecCoreTemplates = {
-  baseUri: templatesDir,
-  templates: content,
-};
+let typeSpecCoreTemplates: LoadedCoreTemplates | undefined;
+export async function getTypeSpecCoreTemplates(host: CompilerHost): Promise<LoadedCoreTemplates> {
```

Comment:
update TypeSpecCoreTemplates to getTypeSpecCoreTemplates so that host can be passed in to read file instead of importing readFile from fs/promises. otherwise, other projects like playground-website or html-program-view will complain about "fs/promises" from vite.